src/native/managed/cdac/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/ComWrappers_1.cs

```
@@ -22,4 +22,72 @@ public TargetPointer GetComWrappersIdentity(TargetPointer address)
         Data.NativeObjectWrapperObject wrapper = _target.ProcessedData.GetOrAdd<Data.NativeObjectWrapperObject>(address);
         return wrapper.ExternalComObject;
     }
+
+    private bool GetComWrappersCCWVTableQIAddress(TargetPointer ccw, out TargetPointer vtable, out TargetPointer qiAddress)
+    {
+        vtable = TargetPointer.Null;
+        qiAddress = TargetPointer.Null;
+        try
+        {
+            TargetPointer vtableAddress = _target.ReadPointer(ccw);
+            if (vtableAddress == TargetPointer.Null)
+                return false;
+            qiAddress = _target.ReadPointer(vtableAddress);
+            if (qiAddress == TargetPointer.Null)
+                return false;
+        }
+        catch (VirtualReadException)
+        {
+            return false;
+        }
```

I'd rather add a new method to target `TryReadPointer` to match the existing `TryRead` so we don't need these try/catches

---

I think that the way it is now works in the large majority of circumstances, as most contract APIs do not handle the exception but let it propagate up to the try...catch in the surrounding cDAC API, which allows the exceptions to be handled in one place. I think if we were to make such an API, it would be rarely used, because writing 
`
if (!TryRead(...)) throw new Exception();
`

is more cumbersome than writing Read(); and letting the VirtualReadException get caught in the cDAC API.

We can write these new methods, I just don't think they are super important or high-priority and they would belong in a separate PR because once you have TryReadPointer you will want TryRead etc.

---

We already have TryRead

---

In this case we also need TryReadCodePointer